sidecar/dkg/dkg.go

```diff
@@ -15,21 +15,26 @@ import (
 	"time"
 )
 
-type DKGCoordinator struct {
+type Protocol interface {
+	RunDKG(identities []crypto.Identity, sessionID []byte, keypair crypto.Keypair) (*dkg.Result, error)
+	ProcessPacket(packet api.SidecarDKGPacket) error
+}
```

Comment:
In Go interfaces usually are best defined on the receiver side.

https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a

Reply:
updated